### PR TITLE
Update the docs of CKB Bitcoin SPV Type Script

### PR DESCRIPTION
**Pull request by KryStar11**
Refinements to the CKB Bitcoin SPV Type Lock Documentation.

### CODE REVIEW

File: contracts/ckb-bitcoin-spv-type-lock/README.md [hunk 1]
```diff
@@ -203,65 +174,55 @@ index of the output SPV info cell, and the proof should be set in
...
+  For the appropriate use of hash type, refer to the "Code Locating" section in [CKB RFC 0022]. `Data1` is
+  introduced in [CKB RFC 0032], and `Data2` is introduced in [CKB RFC 0051].
 
-  So, we define a rule to resolve this case:
+- **Issue #2**: Failed to reorg when there is only 1 stale SPV client.
 
-  - When there is only 1 SPV client is stale, the reorg transaction has to
-    rebuild 2 SPV client.
+  **Solution**: When only one SPV client cell is stale, a typical reorg transaction has the same structure as an update transaction,
+  consisting of one SPV client cell in the inputs and one SPV client cell in the outputs. This similarity can lead to ambiguity.
```
Comment (EthanYuan) [inline contracts/ckb-bitcoin-spv-type-lock/README.md:211]:
```suggestion
  consisting of one SPV client cell in the inputs and one SPV client cell in the outputs. However, this similarity can lead to ambiguity.
```

File: contracts/ckb-bitcoin-spv-type-lock/README.md [hunk 2]
```diff
@@ -203,65 +174,55 @@ index of the output SPV info cell, and the proof should be set in
...
+- **Issue #2**: `VM Internal Error: MemWriteOnExecutablePage`
 
-  When only 1 SPV client is stale, in the normal case, the reorg transaction
-  contains 1 input SPV client cell and 1 output SPV client cell is enough.
+  **Solution**: Don't set hash type[^1] to be `Data`.
 
-  But, the structure of this transaction is totally the same as an update
-  transaction, this leads to ambiguity.
+  For the appropriate use of hash type, refer to the "Code Locating" section in [CKB RFC 0022]. `Data1` is
+  introduced in [CKB RFC 0032], and `Data2` is introduced in [CKB RFC 0051].
```
Comment (EthanYuan) [inline contracts/ckb-bitcoin-spv-type-lock/README.md:206]:
```suggestion
  `Data1` is introduced in [CKB RFC 0032], and `Data2` is introduced in [CKB RFC 0051].
```

Duplicates the link:

> Don't set hash type[^1]

File: contracts/ckb-bitcoin-spv-type-lock/README.md [hunk 3]
```diff
@@ -203,65 +174,55 @@ index of the output SPV info cell, and the proof should be set in
...
-    Since the reorg rarely happens in Bitcoin mainnet, the cost is acceptable.
-
-- Throw "arithmetic operation overflow" when update a Bitcoin SPV instance for
-  a Bitcoin dev chain.
-
-  The bitcoin dev chain doesn't follow the rule of difficulty change.
-
-  So, calculate the next target and partial chain work may lead to an
-  arithmetic operation overflow.
+    - In cases where only one SPV client has failed, the reorg transaction must involve the reconstruction of one separate SPV clients;
```
Comment (EthanYuan) [inline contracts/ckb-bitcoin-spv-type-lock/README.md:215]:
```suggestion
    - In cases where only one SPV client has failed, the reorg transaction must involve the reconstruction of one additional  SPV client;
```

File: contracts/ckb-bitcoin-spv-type-lock/README.md [hunk 4]
```diff
@@ -203,65 +174,55 @@ index of the output SPV info cell, and the proof should be set in
...
+    - Specifically, the reorg transaction for one stale SPV client should include two SPV client cell in the `inputs` and two SPV client
```
Comment (EthanYuan) [inline contracts/ckb-bitcoin-spv-type-lock/README.md:216]:
```suggestion
    Specifically, the reorg transaction for one stale SPV client should include two SPV client cell in the `inputs` and two SPV client
```

File: contracts/ckb-bitcoin-spv-type-lock/README.md [hunk 5]
```diff
@@ -203,65 +174,55 @@ index of the output SPV info cell, and the proof should be set in
...
+    cell in the `outputs` ;
+    - Considering that reorgs are a rare occurrence on the Bitcoin mainnet, the cost incurred by this approach is considered manageable.
```
Comment (EthanYuan) [inline contracts/ckb-bitcoin-spv-type-lock/README.md:218]:
```suggestion
    Considering that reorgs are a rare occurrence on the Bitcoin mainnet, the cost incurred by this approach is considered manageable.
```

**Review by EthanYuan — DISMISSED** [review]

File: contracts/ckb-bitcoin-spv-type-lock/README.md [hunk 6]
```diff
@@ -1,26 +1,21 @@
 
 # CKB Bitcoin SPV Type Lock
```
Comment (EthanYuan) [inline contracts/ckb-bitcoin-spv-type-lock/README.md:2]:
```suggestion
# CKB Bitcoin SPV Type Script
```

Suggest changing the filename as well

File: contracts/ckb-bitcoin-spv-type-lock/README.md [hunk 7]
```diff
@@ -1,26 +1,21 @@
...
 
 ## Brief Introduction
 
-A [Bitcoin] SPV in [CKB] contains a set of cells, this type script is used
-to manage them.
-
-The set is identified by the script [`args`]. The number of live cells with this
-specific `args` script is fixed once created, and all cells in the set are
-destroyed together.
+A Bitcoin  SPV on CKB consists of cells that are managed by the CKB Bitcoin SPV Type Lock and identified by the script args. The number of live cells with the script args remains fixed once created, and these cells will be destroyed collectively as a group.
```
Comment (Flouse) [inline contracts/ckb-bitcoin-spv-type-lock/README.md:8]:
```suggestion
A Bitcoin  SPV on CKB consists of cells that are managed by the CKB Bitcoin SPV Type Script and identified by the script args. The number of live cells with the script args remains fixed once created, and these cells will be destroyed collectively as a group.
```

File: contracts/ckb-bitcoin-spv-type-lock/README.md [hunk 8]
```diff
@@ -35,12 +30,10 @@ There are 2 kinds of cells in a Bitcoin SPV instance:
...
   ```
 
-- Info Cell
-
-  This cell is used to store the basic information of current Bitcoin SPV
-  instance. Such as the ID of the tip client cell.
+- **SPV Info Cell**
 
-  Each Bitcoin SPV instance should contain only 1 info cell.
+  The SPV info cell stores the basic information of the current Bitcoin SPV instance, such as `tip_ client_cell_id`. Each Bitcoin SPV
```
Comment (Flouse) [inline contracts/ckb-bitcoin-spv-type-lock/README.md:35]:
```suggestion
  The SPV info cell stores the basic information of the current Bitcoin SPV instance, such as `tip_client_id`. Each Bitcoin SPV
```

File: contracts/ckb-bitcoin-spv-type-lock/README.md [hunk 9]
```diff
@@ -54,26 +47,16 @@ There are 2 kinds of cells in a Bitcoin SPV instance:
...
 
-  Create all cells for a Bitcoin SPV instance in one transaction.
+- **Create**
 
-  The outputs of this transaction should contain 1 info cell and at least 1 client cell.
+  This operation initiates all necessary cells for a Bitcoin SPV instance in a single transaction.
 
-  In the follow part of this document, we denoted the number of client cells
-  as `n`.
+  The `outputs` include one SPV info cell and at least one SPV client cell. Cells should be consecutive, with the SPV info cell first,
```
Comment (Flouse) [inline contracts/ckb-bitcoin-spv-type-lock/README.md:56]:
```suggestion
  The `outputs` include one SPV info cell and at least three SPV client cell. Cells should be consecutive, with the SPV info cell first,
```

See https://github.com/KryStar11/ckb-bitcoin-spv-contracts/blob/770049914b6818f51b53c11be434a43876b87c45/contracts/ckb-bitcoin-spv-type-lock/src/operations/create.rs#L13-L39

**Review by Flouse — CHANGES_REQUESTED** [review]

File: contracts/ckb-bitcoin-spv-type-lock/README.md [hunk 10]
```diff
@@ -203,65 +177,54 @@ index of the output SPV info cell, and the proof should be set in
...
-- Throw "arithmetic operation overflow" when update a Bitcoin SPV instance for
-  a Bitcoin dev chain.
-
-  The bitcoin dev chain doesn't follow the rule of difficulty change.
-
-  So, calculate the next target and partial chain work may lead to an
-  arithmetic operation overflow.
+    - In cases where only one SPV client has failed, the reorg transaction must involve the reconstruction of one additional SPV client;
+    Specifically, the reorg transaction for one stale SPV client should include two SPV client cell in the `inputs` and two SPV client
+    cell in the `outputs` ;
```
Comment (Flouse) [inline contracts/ckb-bitcoin-spv-type-lock/README.md:219]:
```suggestion
    cells in the `outputs` ;
```

**Review by Flouse — APPROVED** [review]

**Pull request merged**